pkgs/http_profile/lib/http_profile.dart

```diff
@@ -14,20 +192,95 @@ final class HttpClientRequestProfile {
+    _updated();
+  }
+
+  /// The HTTP request method associated with the request.
+  String? get requestMethod => _data['requestMethod'] as String?;
+  set requestMethod(String? value) {
+    _data['requestMethod'] = value;
+    _updated();
+  }
+
+  /// The URI to which the request was sent.
+  String? get requestUri => _data['requestUri'] as String?;
+  set requestUri(String? value) {
+    _data['requestUri'] = value;
+    _updated();
+  }
+
+  /// The time at which the request was initiated, in microseconds.
+  int? get requestStartTimestamp => _data['requestStartTimestamp'] as int?;
+  set requestStartTimestamp(int? value) {
+    _data['requestStartTimestamp'] = value;
+    _updated();
+  }
+
+  /// The time at which the request was completed, in microseconds.
+  int? get requestEndTimestamp => _data['requestEndTimestamp'] as int?;
+  set requestEndTimestamp(int? value) {
+    _data['requestEndTimestamp'] = value;
+    _updated();
+  }
+
+  /// Details about the request.
+  set requestData(HttpProfileRequestData value) {
+    _data['requestData'] = value.toJson();
+    _updated();
+  }
+
+  StreamController<List<int>> _requestBody = StreamController<List<int>>();
+
+  /// The body of the request.
+  StreamSink<List<int>> get requestBodySink {
+    _updated();
+    return _requestBody.sink;
+  }
 
-  HttpClientRequestProfile._();
+  /// Details about the response.
+  set responseData(HttpProfileResponseData value) {
+    _data['responseData'] = value.toJson();
+    _updated();
+  }
+
+  StreamController<List<int>> _responseBody = StreamController<List<int>>();
+
+  /// The body of the response.
+  StreamSink<List<int>> get responseBodySink {
+    _updated();
+    return _responseBody.sink;
+  }
+
+  void _updated() => _data['_lastUpdateTime'] = Timeline.now;
+
+  HttpClientRequestProfile._() {
+    _data['_requestBodyStream'] = _requestBody.stream;
+    _data['_responseBodyStream'] = _responseBody.stream;
+    // This entry is needed to support the updatedSince parameter of
+    // ext.dart.io.getHttpProfile.
+    _data['_lastUpdateTime'] = 0;
+  }
 
-  /// If HTTP profiling is enabled, returns
-  /// a [HttpClientRequestProfile] otherwise returns `null`.
+  /// If HTTP profiling is enabled, returns an [HttpClientRequestProfile],
+  /// otherwise returns `null`.
   static HttpClientRequestProfile? profile() {
-    // Always return `null` in product mode so that the
-    // profiling code can be tree shaken away.
+    // Always return `null` in product mode so that the profiling code can be
+    // tree shaken away.
     if (const bool.fromEnvironment('dart.vm.product') || !profilingEnabled) {
       return null;
     }
     final requestProfile = HttpClientRequestProfile._();
+    addHttpClientProfilingData(requestProfile._data);
+    // This entry is needed to support the id parameter of
+    // ext.dart.io.getHttpProfileRequest.
+    requestProfile._data['id'] =
+        'from_package/${getHttpClientProfilingData().length}';
```

Comment:
Can items ever be removed from `getHttpClientProfilingData()`? We might want to leave the ID assignment to the SDK code if we don't want to expose a `getNextHttpClientProfilingDataId()` like method.

Reply:
Good idea. I've left the ID assignment to the SDK code.